### DISCUSSION TITLE
SUBEDITOR Icons and Labels are inconsistent or outright unusable

### DISCUSSION THREAD

**Discussion by nagginghusband** [FakeFishGames/Barotrauma#13075] [Ideas and suggestions]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

I know this is quite the non-issue since i could mod decals into my game but i want to make subs that only require vanilla package so here are some things id really like to see change. Those are just examples, there are more.

![image](https://github.com/Regalis11/Barotrauma/assets/151722196/64b57eaf-cff3-4626-b348-e9bf2435110b)

Label Icons look clunky and are hard to apply to any design because they look weird, with fine lines and so on. They are also huge and have a really small range to scale.

The basic sign looks great but its already colored, limiting the range to which you can make it fit the design.

Arrows, the Cabins Label and their contemporaries are already lowered alpha and not basic white. Again almost unusable.

### Reproduction steps

pick them from the item selection

### Bug prevalence

Happens regularly

### Single player or multiplayer?

Other

### -

Subeditor

### Version

v1.1.19.3 (Treacherous Tides Hotfix 2)

### Which operating system did you encounter this bug on?

Windows

**Comment by Arctodites** [comment]
I'd second this.  I couldn't count the amount of time I've spent in the sub editor just tweaking sprites to get intelligible signage / wayfinding.  Creating decent custom signage often involves stacking multiple labels and icons, which quickly balloons the overall item / structure count, and the selection of vanilla symbols / icons available in the sub editor is only a subset of the (really cool) icons used by the game itself.  So many of those elements would open up loads of possibilities for vanilla sub design.  In particular: more icons for tools, power sources, monsters (the tentacle icon is my favourite), ships, gun turrets, wayfinding arrows --- all with a sufficient pixel count and in both silhouette and outline versions.  The creative potential of icons could be more than doubled if there were also in-editor options for icon frames and borders.

The other aspect of this is the 'text display' item, whose frame is quite bulky and whose display is limited to printing very few characters at a time.  Although it might be possible to wire up longer synchronised text strings using oscillator components, this is very difficult, and the text displays do not tile seamlessly with each other: one is forced either to use the sprite at full opacity, or to reduce its alpha to zero.  I'd love more options for digital text readouts, as well as scalable (extensible?) backlit monitor screen textures.

It sounds like the devs have lots of higher ambitions for future content (including item editing features) but I think having an expanded vanilla icon pack for custom subs would really elevate the game.

Attached image is just a mock-up of some of my favourite icons that I wish could be added to the sub editor as high-quality sprites.

![Barotrauma_Icons_Examples](https://github.com/Regalis11/Barotrauma/assets/110938168/13103a75-a4d7-4ed6-920a-0b6ad538d0f8)

**Comment by Ek-Videogames** [comment]
As for text - they should really just modernize text label drawing so we have the option to use the higher resolution font of the label letters (with the old letters kept as a legacy option)

I agree that more vanilla label icons would also be good.
